src/TaskLineRenderer.ts

```diff
@@ -140,6 +140,17 @@ async function taskToHtml(
         if (componentString) {
             if (component === 'description') {
                 componentString = GlobalFilter.removeAsWordFromDependingOnSettings(componentString);
+
+                if (renderDetails.layoutOptions?.hideTags) {
+                    // This matches a tag only when it's a complete word
+                    const tagRegExp = RegExp('(#+[a-zA-Z0-9(_)]{1,})', 'ug');
+                    componentString = componentString.replace(tagRegExp, '');
+
+                    const extraSpacesRegExp = RegExp('[ ]{2,}', 'ug');
+                    componentString = componentString.replace(extraSpacesRegExp, ' ');
```

Comment:
There aren't any explicit tests added for the multiple-spaces removal (only implicit ones due to them containing a run of several tests.)

It would be good to see some tests to confirm that this only removes double regular expressions next to the removed tag, and does not adjust other runs of multiple spaces elsewhere in the line...

I tried it out, adding:

```ts
        ['run of      multiple spaces', 'run of      multiple spaces'],
```

And got:

![image](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/4840096/06e19664-3427-4a0d-bd61-cb2694e83a8b)

So I think that this implementation interferes with other parts of the user's line.

Reply:
No idea if it's a good idea, but just in case... another possible implementation of this feature is to iterate over the tags stored in the Task itself.

That would need to be tested with settings:

- a tag-based global filter
- not hiding the global filter

As I suspect that the global filter is excluded from Task.tags.

I suggest this because I think that the code for finding tags has extra cases in it, like not treating text in URLs as tags...

Reply:
> I suggest this because I think that the code for finding tags has extra cases in it, like not treating text in URLs as tags...

Great idea, just what I was looking for. Sorry I didn't figure it out from the beginning =)